tests/resource_name_test.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/resource_name.h>
+#include <aws/testing/aws_test_harness.h>
+
+AWS_TEST_CASE(parse_resource_name_test, s_test_parse_resource_name)
+static int s_test_parse_resource_name(struct aws_allocator *allocator, void *ctx) {
+    (void)allocator;
+    (void)ctx;
+
+    struct aws_byte_cursor arn_string_01 =
+        AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("arn:aws-us-gov:iam::123456789012:user/*");
+    struct aws_resource_name arn_01;
+    AWS_ZERO_STRUCT(arn_01);
+    ASSERT_SUCCESS(aws_resource_name_init_from_cur(&arn_01, &arn_string_01));
+    ASSERT_TRUE(aws_byte_cursor_eq_c_str(&arn_01.partition, "aws-us-gov"));
```

Comment:
We also have AWS_ASSERT_BIN_ARRAY_EQUALS, your call.

Reply:
no, ASSERT_TRUE shouldn't be used if it can be avoided. It makes test output less useful, because you can't see what actually happened.